dialog/file.go

```diff
@@ -480,20 +488,42 @@ func (f *fileDialog) setSelected(file fyne.URI, id int) {
 func (f *fileDialog) setView(view viewLayout) {
 	f.view = view
 	count := func() int {
+		f.dataLock.RLock()
+		defer f.dataLock.RUnlock()
+
 		return len(f.data)
 	}
 	template := func() fyne.CanvasObject {
 		return f.newFileItem(storage.NewFileURI("./tempfile"), true, false)
 	}
 	update := func(id widget.GridWrapItemID, o fyne.CanvasObject) {
+		f.dataLock.RLock()
+
+		if id >= len(f.data) {
+			f.dataLock.RUnlock()
+			return
+		}
```

Comment:
**One more thought here (I'm not necessarily right):**
Can this case ever happen if we lock and unlock the data both before writing and reading?
I'm thinking that if we change the length of the backing data, we have to wait until after the read is done anyway.

Reply:
I just removed this check and was able to reproduce the original issue. I think the happens because `id` is passed in to the function before the lock is obtained. So, I believe it is necessary.